src/stores/WifiSetupStore.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+
+import type { Navigation, WifiNetwork } from '../types';
+
+import { action, computed, observable, reaction } from 'mobx';
+import {
+  WifiConfigureStore,
+  WifiConnectStore,
+} from './ApiRequestStores/SoftApApiStores';
+import SoftApService from '../SoftApService';
+import { LoadObject } from 'brewskey.js-api';
+import { waitForLoaded } from './DAOStores';
+
+type WifiSetupStep = 1 | 2 | 3 | 4;
+
+class WifiSetupStore {
+  @observable wifiSetupStep: WifiSetupStep = 1;
+  @observable particleID: string = '';
+  @observable _wifiSetupLoaderID = null;
+
+  constructor(navigation: Navigation) {
+    reaction(
+      () => this.wifiSetupStep,
+      (wifiSetupStep: WifiSetupStep, currentReaction: Object) => {
+        // todo make replace instead navigate to prevent using hardware back button
+        navigation.navigate(`wifiSetupStep${wifiSetupStep}`);
+        if (wifiSetupStep === 4) {
+          currentReaction.dispose();
+        }
+      },
+    );
+  }
+
+  @computed
+  get wifiSetupLoader(): LoadObject<void> {
+    return this._wifiSetupLoaderID
+      ? WifiConfigureStore.getFromCache(this._wifiSetupLoaderID).map(
+          (): LoadObject<void> => WifiConnectStore.get(),
+        )
+      : LoadObject.empty();
+  }
+
+  @action
+  onStep1Ready = () => {
+    this.setWifiSetupStep(2);
+    this._queryParticleID();
+  };
+
+  @action
+  setParticleID = (particleID: string) => {
+    this.particleID = particleID;
+  };
+
+  @action
+  setWifiSetupStep = (step: WifiSetupStep) => {
+    this.wifiSetupStep = step;
+  };
+
+  @action
+  setupWifi = async (wifiNetwork: WifiNetwork = {}): Promise<void> => {
+    try {
+      this._wifiSetupLoaderID = WifiConfigureStore.fetch(wifiNetwork);
+      await waitForLoaded(() => this.wifiSetupLoader, 15000);
```

Comment:
15 seconds is excessive. Can it just be 5 or even less?

Reply:
this delay doesn't much matter actually, wifiSetupLoader will be filled with error as soon as it comes. this promise is used `await waitForLoaded` to delay navigation transition and timeout should be here should be more than in react-native http request timeout, which seems 10 sec.

Reply:
I'd rather wait for 5 seconds and cancel the request. There should be an API to do that.

Reply:
I don't think that we are able to cancel request with standard promises.

Reply:
https://www.npmjs.com/package/react-native-cancelable-fetch
..but do we really want to add the lib for this?

Reply:
Ahh.. ok, we can skip this then. I guess we can just ignore the request when there is a timeout.  If an error is thrown by the timer but the request resolves the promise will ignore the request.